service.go

```diff
@@ -134,26 +139,40 @@ func (service *SQSService) Start() error {
 		if err != nil {
 			return err
 		}
-	}
 
-	service.setRunning(true)
+		service.started = true
```

Comment:
We could reduce this logic into one variable (`awsSQS`). Instead of setting it at line 114:

```suggestion
		service.aswSQS = awsSQS
```

Then the running method will looks like:

```go
func (service *SQSService) isRunning() bool {
	service.RLock()
	defer service.RUnlock()
	return service.awsSQS != nil
}
```

Reply:
I also agree with this approach.